### DISCUSSION TITLE
Getting an empty table using Angular Datatables server side

### DISCUSSION THREAD

**Discussion by Garry717** [l-lin/angular-datatables#1751] [Q&A]
I'm trying to retrieve data from firebase and display it in the form of Angular DataTable, inorder to implement pagination, sorting utilities. I was able to retrieve the data before using Angular Datatables. I'm fairly new with both Angular and DataTbables. Plz help my to display that data.

I have tried renaming columns data field with all possible combinations, yet can get my data.

https://stackoverflow.com/questions/77144777/getting-an-empty-table-using-angular-datatables-server-side

**Comment by shanmukhateja** [comment]
Your API response is incorrect. If you console.log `resp`, you'll notice the missing parameters in the response.
JSON file doesn't have AJAX callback's mandatory fields like `recordsTotal`, `recordsFiltered`, etc.

Closing this issue as this is issue with server-side implementation.

You will have to fix your API call. Check out the API response in the documentation:

https://l-lin.github.io/angular-datatables/#/basic/new-server-side
